data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -76,6 +80,7 @@ public SqsWorker(final AcknowledgementSetManager acknowledgementSetManager,
         this.acknowledgementSetManager = acknowledgementSetManager;
         this.standardBackoff = backoff;
         this.endToEndAcknowledgementsEnabled = s3SourceConfig.getEndToEndAcknowledgements();
+        this.sqsDynamicVisibilityTimeoutManager = sqsDynamicVisibilityTimeoutManager;
```

Comment:
It would take some decent refactoring so maybe not worth it for now, but this seems like a good opportunity to use the decorator pattern.